Indexer/Indexer.php

```diff
@@ -172,7 +166,7 @@ public function discoverEntity($entity_or_class, EntityManager $em)
         // check if we already saw this type of entity
         // to avoid some expensive work
         if (!array_key_exists($class, self::$indexSettings)) {
-            self::$indexSettings[$class] = $this->getMetaDataLoader()->getMetaData($entity, $em);
+            self::$indexSettings[$class] = $this->loaderChain->getMetaData($entity, $em);
         }
 
         return false !== self::$indexSettings[$class];
```

Comment:
Since the XML and YAML loader return null for classes without mapping information, this method would expose them as being necessary for indexing.